### ISSUE TITLE
Output incompatible with ES3

### ISSUE THREAD

**Issue by neersighted** [rstone770/babelify-external-helpers#1]
For those of us who use Babel with older browsers, such as IE8, the output is incompatible as it uses reserved keywords (`default`). It is simple to quote these strings in order to make them valid in old versions of JScript.

This plugin does not pass through browserify transforms it seems (`es3ify` does not work), nor does it get transformed by plugins specified in a `.babelrc`, preventing it from being used with IE8.

**Comment by rstone770** [comment]
I'm having trouble understanding the issue here. From what i understand, it seems like the issue you are having has to do with browserify, babelify, or es3ify. The external helpers plugin does not actually process the output except to inject `usedHelpers` and does not honor any configurations in babelrc since it does not use them. If you can add some more information like input and expected output, I can look into this further.

**Comment by neersighted** [comment]
To clarify, it would be nice if this could somehow be a transform so that it would respect the transform stack and could be mangled into ES3. If not, it can be worked around by post-processing the bundle I suppose.

**Comment by rstone770** [comment]
I ran some tests and see the issues you are having.

For example.

```
  babelHelpers.interopRequireDefault = function (obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  };
```

Unfortunately these helpers are generated by babel [here](https://github.com/babel/babel/blob/8cac20d845c8d20dafce6937ff374b3b0d948e20/packages/babel-helpers/src/helpers.js#L238)... and the module source is injected into the pipeline AFTER the transforms have taken place. This is definitely an issue, and ill take a look at it. 

Feel free to give it a stab, if you have the time.

**Comment by rstone770** [comment]
I have a couple solutions in mind and none of them will offer 100% interoperability of all transforms and plugins. Curious of what you think.

The first solution is to offer a `compatibility` mode option that will just transform the module into es3. This only solves this one issue, and other plugins which transform sources will not be supported. They will work, but just like es3ify, will not be aware of the module and will not be able to transform it.

Another solution would be to 'defer' module transforms for the helper module until the helper source is built. After module has been complete i can 'play' the transforms over the module as its being minted. I'm not sure of the side effects this could cause which makes me anxious. Although it would solve a wider class of problems.

**Comment by neersighted** [comment]
Compatibility would be the easy way out. However, bundling before transforms are applied would eb the best way.

**Comment by rstone770** [comment]
This is doable, and an idea that i've been toying with and probably will
just implement for the time being. I would really like plugin to bundle the
minimal amount of runtime that's needed. The current way this works is it
listens to babelify as files are being transformed one by one and collects
metadata about the used helpers. After the list is complete,ie the
transforms are complete it generates the source module for the run time
helpers and adds it to the bundle. The issue obviously is generating all
the helpers needed before the transforms complete.

The best solution for the time being, is to skip the optimal module
construction and inject the entire runtime into the bundle. The file size
will obviously be bigger but this guarantees compatibility until i figure
something else out. Compatibility is much more important than saving a
couple kb on the file size which will be negligible after minification and
or gzip.

In the meantime i'll read through browserify more thoroughly to see if I
can use a different route of attack to get best of both worlds.

What do you think?

**Comment by neersighted** [comment]
Sounds good to me, and still better than helpers-per-module. I agree it would be very nice to bundle only what is needed, but that means throwing browser support out the window.

**Comment by rstone770** [comment]
I just released v1.1.0 which includes both #1 and #2.

**Comment by neersighted** [comment]
Great, thanks!
